pkg/cmd/step/git/credentials/step_git_credentials.go

```diff
@@ -108,13 +107,12 @@ func (o *StepGitCredentialsOptions) Run() error {
 			return errors.Wrapf(err, "failed to find secret '%s' in namespace '%s'", o.CredentialsSecret, ns)
 		}
 
-		creds := credentials{
-			user:       string(secret.Data["user"]),
-			password:   string(secret.Data["token"]),
-			serviceURL: string(secret.Data["url"]),
+		creds, err := credentialhelper.CreateGitCredentialFromURL(string(secret.Data["url"]), string(secret.Data["token"]), string(secret.Data["user"]))
```

Comment:
You might be talking to the wrong person. @garethjevans  introduced this in e9aa7f3840da37c6e706856785e2b1759867b749. I just rebased on top of this change here and made it work with the changes I am making. I actually meant to ask you whether you have seen this change since I was also wondering why we would be reading directly from the secret. @garethjevans might have an answer to this.

@ccojocar is there an easy fix to this? Do you want me to address this as part of this PR?

Reply:
If we want to have this working in all setup combinations, we need to switch to `AuthConfigService`. Actually, there is a `GitAuthConfigService` in the `CommonOptions` which loads the auth config and abstracts all the secrets lookup based on the secret source and the setup available. In the latest boot version, that secret is not anymore present. There is a config map with URI references into the vault.

Reply:
What I am confused about is that we are using already a AuthConfigService, at least for the code which existed. I don't understand why we are reading this secret directly in this case.

Reply:
Looking closer to that code and it seems that the loading directly from secrets is activated only when there is some value in the `JX_CREDENTIALS_FROM_SECRET`.  I think we can ignore my comment for now.

Reply:
works for me :-)